src/serac/physics/heat_transfer.hpp

```diff
@@ -986,6 +1027,12 @@ class HeatTransfer<order, dim, Parameters<parameter_space...>, std::integer_sequ
   /// The temperature finite element state
   serac::FiniteElementState temperature_;
 
+  /// An optional vector of computed temperatures used for dynamic adjoints
+  std::vector<serac::FiniteElementState> temperatures_;
```

Comment:
temperature_ and temperatures_ look very similar.  Perhaps OK, but maybe consider making the name more distinct, e.g. checkpointed_temperatures_?